openassessment/assessment/models/peer.py

```diff
@@ -248,6 +255,8 @@ def get_waiting_step_details(cls, course_id, item_id, submission_uuids):
         ).annotate(
             graded_count=models.Count('graded'),
             graded_by_count=models.Count('graded_by'),
+        ).filter(
+            graded_by_count__lt=must_be_graded_by
```

Comment:
@jnlapierre This is the main backend change from the product review. This hides students that already received the required number of peer assessments from the waiting step dashboard since `grading_completed_at__isnull` doesn't always indicate that.